03-Modify-Raster.Rmd

```diff
@@ -74,197 +72,240 @@ Read more about the `raster` package in R.</a>
+4. Bonus challenge: Change the color of and add a title to the plot you created 
+in question 3. 
+
+```{r challenge-code-1 }
+#1) Depends on types of trees.  0m to <50m?
+#2) Looks at histogram or min/max values. 
+minValue(CHM)
+maxValue(CHM)
+#3) 
+hist(CHM, maxpixels=ncell(CHM))
+
+#4)
 hist(CHM, 
      col = "purple",
-     main = "Histogram of NEON Canopy Height Model\n Harvard Forest")
+     main = "Histogram of NEON Canopy Height Model\n Harvard Forest",
+     maxpixels=ncell(CHM))
 
 ```
 
-#Raster Overlay
-
-Raster math is an appropriate aporoach to this calculation if
+#Efficient Raster Math: Overlay Function
+Basic raster math like we just did, is an appropriate aporoach to a calculation 
+if
 
 1. The rasters we are using are small in size.
 2. The calculations we are performing are simple.
 
-Raster math is easy to code up, however it is less efficient as computation becomes
-more complex or as file sizes become larger. To deal with larger rasters that are
-stored as individual files we should use the `overlay` function
+Raster math is intuitive and easy to code, however it is less efficient as 
+computation becomes more complex or as file sizes become larger. The `overlay()`
+function is more efficient when
 
-NOTE: If our rasters are stacked and stored as `RasterStack` or `RasterBrick`
-objects in R, then we can use `calc`.
-{: .notice }
+1. The rasters we are using are larger in size. 
+2. The rasters are stored as individual files. 
+3. The computations needed are complex. 
 
-Let's have a look at how the `overlay` function works. The `overlay` function 
-takes two or more rasters and applies a function to them using efficient processing
-methods. The syntax looks like:
+>NOTE: If the rasters are stacked and stored as `RasterStack` or `RasterBrick`
+>objects in R, then we should use `calc()`. `overlay()` will not work on stacked
+>rasters. 
 
-`outputRaster <- overlay(raster1, raster2, functionName )`
+The `overlay()` function takes two or more rasters and applies a function to
+them using efficient processing methods. The syntax is
 
-##Writing Functions in R
-
-A function represents a set of tasks that need to be repeated over and over. 
-The syntax for a function in R is as follows:
+`outputRaster <- overlay(raster1, raster2, fun=functionName)`
 
-`functionName <- function(variable1, variable2){
+##Writing Functions in R
+Before we can continue with `overlay()` we need to discuss writing functions. A 
+function represents a set of tasks that need to be repeated over and over. 
 
+LINK TO MORE ABOUT FUNCTIONS IN R 
+I like explanations (how to AND why) here: http://nicercode.github.io/guides/functions/
+but I'm hesitant to link to blogs that might disappear so also QuickR link:
+http://www.statmethods.net/management/userfunctions.html
+Thoughts? 
 
-}
+A simplified syntax for a function in R is as follows:
 
-LINK TO MORE ABOUT FUNCTIONS IN R 
+`functionName <- function(variable1, variable2){WhatYouWantDone, WhatToReturn}`
```

Comment:
@mjones01 it was suggested by the group that maybe instead of writing another function above - to just include the function in the actual line of overlay code. thoughts?